### PR TITLE
Request to add fwbackups (com.diffingo.fwbackups)

### PR DESCRIPTION
**Pull request by stewartadam**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - filesystem=host: necessary because the application is a backup program, and users expect files to be backed up and restored on their original paths. This also preserves back-compat with users of the program prior to switching to flatpak distribution.
  - socket=ssh-auth: the application supports sending files to remote hosts over SFTP, and should be able to leverage any configured SSH identities
  - talk-name=org.freedesktop.Flatpak: backup and restore processes execute various commands on the host (tar, rsync, dpkg, rpm) to achieve the backup and restore functionality, and doing so within the sandbox is not feasible.
  - unset-env=LD_PRELOAD: it's fine if this exists on the host but we don't want this passed into the sandbox.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://diffingo.com/oss/fwbackups || https://github.com/stewartadam/fwbackups
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.diffingo.fwbackups.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  "finish-args": [
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--socket=wayland",
+    "--filesystem=host",
+    "--share=network",
+    "--socket=ssh-auth",
+    "--talk-name=org.freedesktop.Flatpak",
+    "--unset-env=LD_PRELOAD"
```
Comment (hfiguiere) [inline com.diffingo.fwbackups.json:21]:
```suggestion
    "--talk-name=org.freedesktop.Flatpak"
```

it's uncessary

File: com.diffingo.fwbackups.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+  "app-id": "com.diffingo.fwbackups",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "43",
+  "sdk": "org.gnome.Sdk",
+  "command": "fwbackups",
+  "build-options": {
+    "build-args": [
+      "--share=network"
```
Comment (hfiguiere) [inline com.diffingo.fwbackups.json:9]: build must run off-line. This is not allowed.

Comment (stewartadam) [reply com.diffingo.fwbackups.json]: Thanks for the fast review -- reworked the installation to source wheels from pypi with reqs2flatpak. Let me know how it looks now.

**Comment by barthalion** [comment]
> talk-name=org.freedesktop.Flatpak: backup and restore processes execute various commands on the host (tar, rsync, dpkg, rpm) to achieve the backup and restore functionality, and doing so within the sandbox is not feasible.

To be fair, there's probably no reason why the app couldn't bundle some or even all of these tools. It's not a blocker, though.

**Comment by stewartadam** [comment]
Build failed but looks to be on linting (due to the `--talk-name` I think) -- are any changes required?

**Comment by JakobDev** [comment]
> Build failed but looks to be on linting (due to the --talk-name I think) -- are any changes required?

Make a PR against [this file](https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/staticfiles/exceptions.json) and add a exception for your App

**Comment by barthalion** [comment]
bot, build

**Comment by barthalion** [comment]
bot, build com.diffingo.fwbackups

**Pull request closed without merging**